docs/modules/ROOT/pages/documentation/payara-server/server-configuration/certificate-management.adoc

```diff
@@ -0,0 +1,490 @@
+|The name of the instance to get the certificate from.
+|server
+|no
+
+|`--listener`
+|String
+|The name of the HTTP or IIOP listener to get the certificate from.
+|N/A
+|no
+
+|`--domain_name`
+|String
+|The name of the domain where the target instance exists.
+|domain1
+|no
+
+|`--domaindir`
+|String
+|The path to the directory containing the target domain.
+|${installDir}/glassfish/domains
+|no
+
+|`--alias` (primary)
+|String
+|The alias name of the certificate to generate a CSR for.
+|N/A
+|yes
+
+|===
+
+[[example]]
+=== Example
+
+[source, shell]
+----
+asadmin> generate-csr --listener http-listener-2 --target Instance1 test_cert
+----
+
+[[add-to-keystore]]
+== `add-to-keystore`
+
+*Usage*::
+`asadmin> add-to-keystore --target=instancename --listener=listenername --file /path/to/file alias`
+
+*Aim*::
+This command adds a certificate bundle (e.g. .p12 or .jks file) or certificate (e.g. .cert file) to the target instance
+or listener's key store using the provided alias.
+
+If the instance or listener is configured to use the default key store, the command will instead synchronise
+the instance with the DAS (under the assumption the certificate has been added to the default key store of
+the DAS), since any certificates added to the instance stores would be lost upon next synchronisation.
+
+*Note*::
+This will overwrite an entry already present with the same alias.
+
+[[command-options]]
+=== Command Options
+
+[cols="3,1,5,1,1",options="header"]
+|===
+|Option
+|Type
+|Description
+|Default
+|Mandatory
+
+|`--target`
+|String
+|The name of the instance to add the certificate to.
+|server
+|no
+
+|`--listener`
+|String
+|The name of the HTTP or IIOP listener to add the certificate to.
+|N/A
+|no
+
+|`--domain_name`
+|String
+|The name of the domain where the target instance exists.
+|domain1
+|no
+
+|`--domaindir`
+|String
+|The path to the directory containing the target domain.
+|${installDir}/glassfish/domains
+|no
+
+|`--file`
+|File
+|The certificate or bundle file to add to the target key store
+|N/A
+|yes
+
+|`--alias` (primary)
+|String
+|The alias name to store the certificate or bundle the key store under.
+|N/A
+|yes
+
+|===
+
+[[example]]
+=== Example
+
+[source, shell]
+----
+asadmin> add-to-keystore --file /home/anon/Downloads/mycert.p12 mycert
+----
+
+[[add-to-truststore]]
+== `add-to-truststore`
+
+*Usage*::
+`asadmin> add-to-truststore --target=instancename --listener=listenername --file /path/to/file alias`
+
+*Aim*::
+This command adds a certificate bundle (e.g. .p12 or .jks file) or certificate (e.g. .cert file) to the target instance
+or listener's trust store using the provided alias.
+
+If the instance or listener is configured to use the default trust store, the command will instead synchronise
+the instance with the DAS (under the assumption the certificate has been added to the default trust store of
+the DAS), since any certificates added to the instance stores would be lost upon next synchronisation.
```

Comment:
```suggestion
If the instance or listener is configured to use the default trust store, the command will place the certificate in the default trust store of the DAS and then synchronise
the instance with the DAS. This is to preserve the certificates across instance restarts because any certificates added directly to the instance stores would be lost upon next synchronisation.
```

Reply:
See comments elsewhere, your suggestion isn't what happens.